qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -466,6 +466,9 @@ def __init__(self, graph, source_basis, target_basis):
         self._num_gates_remain_for_rule = {}
         save_index = -1
         for edata in self.graph.edges():
+            # We only care about edges with data.
+            if edata is None:
+                continue
```

Comment:
I think this might be masking potential logic problems; at the time that this initialisation runs, all edges should have data.